numbat/src/typechecker.rs

```diff
@@ -1019,12 +1019,15 @@ impl TypeChecker {
                     .iter()
                     .map(|p| match p {
                         StringPart::Fixed(s) => Ok(typed_ast::StringPart::Fixed(s.clone())),
-                        StringPart::Interpolation(span, expr) => {
-                            Ok(typed_ast::StringPart::Interpolation(
-                                *span,
-                                Box::new(self.check_expression(expr)?),
-                            ))
-                        }
+                        StringPart::Interpolation {
+                            span,
+                            expr,
+                            specifiers,
+                        } => Ok(typed_ast::StringPart::Interpolation {
+                            span: *span,
+                            specifiers: specifiers.clone(),
+                            expr: Box::new(self.check_expression(expr)?),
+                        }),
```

Comment:
Would be nice to be able to format them to be honest, for example to add whitespace padding or similar things. Not sure if `strfmt` supports that though.

Reply:
Right, it should work great for strings already. And things that are not number or a string... we can maybe convert them to a string for now.

Reply:
For dates, we could even think about supporting things like `"{now():%Y-%m-%d}"`. That shouldn't be too hard. We could use the type checker to distringuish those cases from everything else. And then call `format_datetime` (in the VM) instead of `strfmt::strfmt`.